doc/developers/developing_scikit.rst

```diff
@@ -0,0 +1,761 @@
+
+Attributes that have been estimated from the data must always have a name
+ending with trailing underscore, for example the coefficients of
+some regression estimator would be stored in a ``coef_`` attribute after
+``fit`` has been called.
+
+The estimated attributes are expected to be overridden when you call ``fit``
+a second time.
+
+Optional Arguments
+^^^^^^^^^^^^^^^^^^
+
+In iterative algorithms, the number of iterations should be specified by
+an integer called ``n_iter``.
+
+Pairwise Attributes
+^^^^^^^^^^^^^^^^^^^
+
+An estimator that accept ``X`` of shape ``(n_samples, n_samples)`` and defines
+a :term:`_pairwise` property equal to ``True`` allows for cross-validation of
+the dataset, e.g. when ``X`` is a precomputed kernel matrix. Specifically,
+the :term:`_pairwise` property is used by ``utils.metaestimators._safe_split``
+to slice rows and columns.
+
+.. _testing_coverage:
+
+Testing and improving test coverage
+===================================
+
+High-quality `unit testing <https://en.wikipedia.org/wiki/Unit_testing>`_
+is a corner-stone of the scikit-learn development process. For this
+purpose, we use the `pytest <https://docs.pytest.org>`_
+package. The tests are functions appropriately named, located in `tests`
+subdirectories, that check the validity of the algorithms and the
+different options of the code.
+
+The full scikit-learn tests can be run using 'make' in the root folder.
+Alternatively, running 'pytest' in a folder will run all the tests of
+the corresponding subpackages.
+
+We expect code coverage of new features to be at least around 90%.
+
+For guidelines on how to use ``pytest`` efficiently, see the
+:ref:`pytest_tips`.
+
+Writing matplotlib related tests
+--------------------------------
+
+Test fixtures ensure that a set of tests will be executing with the appropriate
+initialization and cleanup. The scikit-learn test suite implements a fixture
+which can be used with ``matplotlib``.
+
+``pyplot``
+    The ``pyplot`` fixture should be used when a test function is dealing with
+    ``matplotlib``. ``matplotlib`` is a soft dependency and is not required.
+    This fixture is in charge of skipping the tests if ``matplotlib`` is not
+    installed. In addition, figures created during the tests will be
+    automatically closed once the test function has been executed.
+
+To use this fixture in a test function, one needs to pass it as an
+argument::
+
+    def test_requiring_mpl_fixture(pyplot):
+        # you can now safely use matplotlib
+
+Workflow to improve test coverage
+---------------------------------
+
+To test code coverage, you need to install the `coverage
+<https://pypi.org/project/coverage/>`_ package in addition to pytest.
+
+1. Run 'make test-coverage'. The output lists for each file the line
+    numbers that are not tested.
+
+2. Find a low hanging fruit, looking at which lines are not tested,
+    write or adapt a test specifically for these lines.
+
+3. Loop.
+
+
+Issue Tracker Tags
```

Comment:
I'm pretty sure this belongs in the contributer guide, not here: it is about the specific development processes for the scikit-learn project